making_change/making_change.py

```diff
@@ -3,6 +3,14 @@
 import sys
 
 def making_change(amount, denominations):
```

Comment:
This looks good.  I wasn't able to get this to run the large test.  Think about using a `[]` or `{}` as a cache and some loops